src/cogent3/core/new_alignment.py

```diff
@@ -355,8 +364,10 @@ def to_alphabet(
         old = self.alphabet.as_bytes()
         new = alphabet.as_bytes()
         for seqid in self.names:
+            # refactor: design
+            # this conversion needs to be a method on the alphabet
+            # so it can be used by Sequence.to_moltype
             seq_data = self.get_seq_array(seqid=seqid)
-
             convert_old_to_bytes = new_alpha.array_to_bytes(old)
```

Comment:
since they're constant it's worth moving construction of those converters outside the loop